frontend/packages/operator-lifecycle-manager/src/components/create-operand.tsx

```diff
@@ -68,32 +114,68 @@ enum Validations {
+/*
+ * Determines if a field contains a descriptor in it's capabilities. If only prefix is
+ * provided, this will return true if the field has ANY capability that starts with the prefix. If
+ * both prefix and suffix are provided, this will return true only if the field has a capability
+ * that matches the concatenation of prefix + suffix.
  */
-type OperandField = {
-  path: string;
-  displayName: string;
-  description?: string;
-  type: JSONSchema6TypeName;
-  required: boolean;
-  validation: {
-    [Validations.maximum]?: number;
-    [Validations.minimum]?: number;
-    [Validations.maxLength]?: number;
-    [Validations.minLength]?: number;
-    [Validations.pattern]?: string;
-  };
-  capabilities: (SpecCapability | StatusCapability)[];
+const hasDescriptor = (field, prefix, suffix = null) => {
+  if (suffix) {
+    return _.includes(field.capabilities, `${prefix}${suffix}`);
+  }
+
+  return _.some(field.capabilities, (capability) => capability.startsWith(prefix));
 };
 
-type FieldErrors = {
-  [path: string]: string;
+/*
+ * Accepts an OperandField and returns a name and group type if that field is either a
+ * a field group or array field group. Returned as object to allow destructuring of needed values
+ * only.
+ */
+const parseGroupDescriptor = (
+  field: OperandField,
+): { groupName?: string; groupType?: string; match?: string } => {
+  const groupDescriptor = _.find(
+    field.capabilities,
+    (descriptor) =>
+      descriptor.startsWith(SpecCapability.fieldGroup) ||
+      descriptor.startsWith(SpecCapability.arrayFieldGroup),
+  );
+  const [match, groupType, groupName] = groupDescriptor.match(GROUP_PATTERN) || [];
+  return { match, groupName, groupType };
+};
+
+/*
+ * Splits a path string into path before the array index, the array index itself, and the path after
+ * the index. Returns as object to allow destructuring of needed values only.
+ */
+const parseArrayPath = (path: string) => {
+  const [match, pathBeforeIndex, index, pathAfterIndex] = path.match(ARRAY_INDEX_PATTERN) || [];
+  return match ? { index: _.parseInt(index), match, pathBeforeIndex, pathAfterIndex } : { match };
 };
 
-const defaultValueFor = (field: OperandField) => {
+/*
+ * If the path contains an array index, this function will return the path with the 'operation'
+ * callback return value in place of the existing array index. If no array index is in the path,
+ * the original path is returned.
+ */
+const modifyArrayFieldPathIndex = (
+  path: string,
+  operation: (index?: number) => string | number,
+): string => {
+  const { match, index, pathBeforeIndex, pathAfterIndex } = parseArrayPath(path);
+  if (!match) {
+    return path;
+  }
+  return `${pathBeforeIndex}[${operation(index)}]${pathAfterIndex && `.${pathAfterIndex}`}`;
+};
+
+// Accepts a SpecCapbability[] array and returns an appropriate default value for that field
+const defaultValueFor = (capabilities: OperandField['capabilities']) => {
```

Comment:
Nothing to change now (not an issue with this PR), but getting this far and seeing how big this file is makes me think it needs its own directory & some code splitting to make it easier to maintain.